rustls/src/msgs/message.rs

```diff
@@ -435,36 +465,65 @@ impl<'a> TryFrom<BorrowedPlainMessage<'a>> for Message<'a> {
 /// This type also cannot decode its internals and
 /// cannot be read/encoded; only `OpaqueMessage` can do that.
 #[derive(Debug)]
-pub struct BorrowedPlainMessage<'a> {
+pub struct BorrowedPlainMessageInbound<'a> {
     pub typ: ContentType,
     pub version: ProtocolVersion,
-    pub payload: BorrowedPlainPayload<'a>,
+    pub payload: &'a [u8],
 }
 
-impl<'a> BorrowedPlainMessage<'a> {
-    pub fn to_unencrypted_opaque(&self) -> OpaqueMessage {
-        OpaqueMessage {
-            version: self.version,
-            typ: self.typ,
-            payload: Payload::Owned(self.payload.to_vec()),
-        }
+impl BorrowedPlainMessage for BorrowedPlainMessageInbound<'_> {
+    fn typ(&self) -> ContentType {
+        self.typ
+    }
+    fn version(&self) -> ProtocolVersion {
+        self.version
+    }
+    fn payload_to_vec(&self) -> Vec<u8> {
+        self.payload.to_vec()
+    }
+    fn payload_len(&self) -> usize {
+        self.payload.len()
     }
 
-    pub fn encoded_len(&self, record_layer: &RecordLayer) -> usize {
-        OpaqueMessage::HEADER_SIZE as usize + record_layer.encrypted_len(self.payload.len())
+    fn is_ccs(&self) -> bool {
+        self.typ == ContentType::ChangeCipherSpec && self.payload.len() == 1 && self.payload[0] == 1
     }
+}
 
-    pub fn into_owned(self) -> PlainMessage {
-        let Self {
-            typ,
-            version,
-            payload,
-        } = self;
-        PlainMessage {
-            typ,
-            version,
-            payload: Payload::new(payload),
-        }
+/// A TLS frame, named TLSPlaintext in the standard.
+///
+/// This type differs from `OpaqueMessage` because it borrows
+/// its payload.  You can make a `OpaqueMessage` from an
+/// `BorrowMessage`, but this involves a copy.
+///
+/// This type also cannot decode its internals and
+/// cannot be read/encoded; only `OpaqueMessage` can do that.
+#[derive(Debug)]
+pub struct BorrowedPlainMessageOutbound<'a> {
+    pub typ: ContentType,
+    pub version: ProtocolVersion,
+    pub payload: BorrowedPlainPayload<'a>,
+}
+
+impl BorrowedPlainMessage for BorrowedPlainMessageOutbound<'_> {
+    fn typ(&self) -> ContentType {
+        self.typ
+    }
+    fn version(&self) -> ProtocolVersion {
```

Comment:
same nit here w.r.t whitespace

Reply:
(I think this nit is still unaddressed - not a blocker, just calling it out in case it was accidental)